src/OpenSage.Game/Data/Utilities/Extensions/BinaryReaderExtensions.cs

```diff
@@ -275,6 +277,30 @@ public static IndexedTriangle ReadIndexedTri(this BinaryReader reader)
                 reader.ReadUInt16());
         }
 
+        public static ButtonRecord ReadButtonRecord(this BinaryReader reader)
```

Comment:
This method, and `ReadButtonAction`, should be static `Parse` methods on `ButtonRecord` and `ButtonAction` respectively, because they're specific to .apt.